### PR TITLE
Deduplicate client ready checks, fix a few places they were missing

### PR DESCRIPTION
**Pull request by jshufro**

### CODE REVIEW

File: rocketpool-cli/network/generate-tree.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ const (
 func generateRewardsTree(c *cli.Context) error {
 
 	// Get RP client
-	rp, err := rocketpool.NewClientFromCtx(c)
```
Comment (jshufro) [inline rocketpool-cli/network/generate-tree.go:21]: this was missing the ready check

File: rocketpool-cli/node/distributor.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ import (
 func initializeFeeDistributor(c *cli.Context) error {
 
 	// Get RP client
-	rp, err := rocketpool.NewClientFromCtx(c)
```
Comment (jshufro) [inline rocketpool-cli/node/distributor.go:17]: was missing here as well

File: rocketpool-cli/node/distributor.go [hunk 3]
```diff
@@ -73,7 +73,7 @@ func initializeFeeDistributor(c *cli.Context) error {
 func distribute(c *cli.Context) error {
 
 	// Get RP client
-	rp, err := rocketpool.NewClientFromCtx(c)
```
Comment (jshufro) [inline rocketpool-cli/node/distributor.go:76]: was missing here as well

File: rocketpool-cli/node/send-message.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ import (
 func sendMessage(c *cli.Context, toAddressOrENS string, message []byte) error {
 
 	// Get RP client
-	rp, err := rocketpool.NewClientFromCtx(c)
```
Comment (jshufro) [inline rocketpool-cli/node/send-message.go:17]: was missing here as well

File: rocketpool-cli/node/sign-message.go [hunk 5]
```diff
@@ -24,7 +24,7 @@ type PersonalSignature struct {
 func signMessage(c *cli.Context) error {
 
 	// Get RP client
-	rp, err := rocketpool.NewClientFromCtx(c)
```
Comment (jshufro) [inline rocketpool-cli/node/sign-message.go:27]: doesn't need synced clients :)

File: rocketpool-cli/node/sync.go [hunk 6]
```diff
@@ -50,7 +50,7 @@ func printSyncProgress(status *api.ClientManagerStatus, name string) {
 func getSyncProgress(c *cli.Context) error {
 
 	// Get RP client
-	rp, err := rocketpool.NewClientFromCtx(c)
```
Comment (jshufro) [inline rocketpool-cli/node/sync.go:53]: this command cares whether the clients are synced, but handles it further down, of course.

File: rocketpool-cli/node/withdrawal-address.go [hunk 7]
```diff
@@ -158,7 +152,7 @@ func setWithdrawalAddress(c *cli.Context, withdrawalAddressOrENS string) error {
 func confirmWithdrawalAddress(c *cli.Context) error {
 
 	// Get RP client
-	rp, err := rocketpool.NewClientFromCtx(c)
```
Comment (jshufro) [inline rocketpool-cli/node/withdrawal-address.go:161]: was missing ready checks

File: rocketpool-cli/service/service.go
Comment (jshufro) [inline rocketpool-cli/service/service.go:1]: this whole file doesn't care whether the clients are synced, so it ignores the bool

File: rocketpool-cli/wallet/ens-name.go [hunk 8]
```diff
@@ -12,7 +12,7 @@ import (
 func setEnsName(c *cli.Context, name string) error {
 
 	// Get RP client
-	rp, err := rocketpool.NewClientFromCtx(c)
```
Comment (jshufro) [inline rocketpool-cli/wallet/ens-name.go:15]: was missing here as well

File: rocketpool-cli/wallet/export.go [hunk 9]
```diff
@@ -13,7 +13,7 @@ import (
 func exportWallet(c *cli.Context) error {
 
 	// Get RP client
-	rp, err := rocketpool.NewClientFromCtx(c)
```
Comment (jshufro) [inline rocketpool-cli/wallet/export.go:16]: doesn't care about client status

File: rocketpool-cli/wallet/init.go [hunk 10]
```diff
@@ -13,7 +13,7 @@ import (
 func initWallet(c *cli.Context) error {
 
 	// Get RP client
-	rp, err := rocketpool.NewClientFromCtx(c)
```
Comment (jshufro) [inline rocketpool-cli/wallet/init.go:16]: doesn't care about client status

File: rocketpool-cli/wallet/purge.go [hunk 11]
```diff
@@ -12,7 +12,7 @@ import (
 func purge(c *cli.Context) error {
 
 	// Get RP client
-	rp, err := rocketpool.NewClientFromCtx(c)
```
Comment (jshufro) [inline rocketpool-cli/wallet/purge.go:15]: doesn't care about client status

File: rocketpool-cli/wallet/recover.go
Comment (jshufro) [inline rocketpool-cli/wallet/recover.go:1]: in recover/test-recovery, we only care about client status when -k isn't passed, so we record the result in `ready` and check it

File: rocketpool-cli/wallet/status.go [hunk 12]
```diff
@@ -12,7 +12,7 @@ import (
 func getStatus(c *cli.Context) error {
 
 	// Get RP client
-	rp, err := rocketpool.NewClientFromCtx(c)
```
Comment (jshufro) [inline rocketpool-cli/wallet/status.go:15]: doesn't care about client status

File: shared/services/rocketpool/client.go [hunk 13]
```diff
@@ -62,6 +64,13 @@ const (
 	DebugColor = color.FgYellow
 )
 
+// When printing sync percents, we should avoid printing 100%.
+// This function is only called if we're still syncing,
+// and the `%0.2f` token will round up if we're above 99.99%.
+func SyncRatioToPercent(in float64) float64 {
```
Comment (jshufro) [inline shared/services/rocketpool/client.go:70]: might make sense to move this to a different util file, but for now i shoved it here

**Comment by jclapis** [comment]
Good addition, this is definitely a better way to do it. Thanks Patches!

**Pull request merged**